src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -200,48 +220,68 @@ protected override NuGetIndexSearcher CreateSearcher(IndexReader reader)
-                filterCreationProcessor.AddHandler(h00);
-                filterCreationProcessor.AddHandler(h01);
-                filterCreationProcessor.AddHandler(h10);
-                filterCreationProcessor.AddHandler(h11);
-                filterCreationProcessor.AddHandler(curatedFeedHandler);
-
-                filterCreationProcessor.Process(combined);
-
-                cruratedfeeds.Clear();
-
-                var latest = new Filter[][] { new Filter[] { h00.Result, h01.Result }, new Filter[] { h10.Result, h11.Result } };
-
-                var latestBitSet = BitSetCollector.CreateBitSet(combined, latest[0][1]);
-                var latestStableBitSet = BitSetCollector.CreateBitSet(combined, latest[0][0]);
+                // We want to be able to filter on curated feeds as well
+                var curatedFeedHandler = new CuratedFeedHandler(_curatedFeeds);
+                indexReaderProcessor.AddHandler(curatedFeedHandler, skipDeletes: true, requiresPerSegmentDocumentNumber: true);
 
+                // Traverse the index and segments
+                indexReaderProcessor.Process(reader);
+                
+                // Set filters
+                var latest = new Filter[][]
+                {
+                    new Filter[] { h00.Result, h01.Result },
+                    new Filter[] { h10.Result, h11.Result }
+                };
+                
+                var latestBitSet = BitSetCollector.CreateBitSet(reader, latest[0][1]);
+                var latestStableBitSet = BitSetCollector.CreateBitSet(reader, latest[0][0]);
+                
+                // Done loading index
+                _logger.LogInformation("Original {MaxDoc} (deletes: {NumDeletedDocs})", reader.MaxDoc, reader.NumDeletedDocs);
+                
                 // The point of having a specific subclass of the IndexSearcher is that we want to associate a bunch of auxilliary data along
                 // with that specific instance of the reader. The lifetimes are assocaited, hense the inheritance relationship.
 
                 _logger.LogInformation("About to create a new NuGetIndexSearcher");
-
+                
                 // Create a NuGetIndexSearcher
                 return new NuGetIndexSearcher(
                     this,
-                    combined,
                     reader,
                     reader.CommitUserData,
                     curatedFeedHandler.Result,
                     latest,
                     versionsHandler.Result,
-                    rankings,
+                    rankingsHandler.Result,
                     latestBitSet,
-                    latestStableBitSet);
+                    latestStableBitSet,
+                    ownersHandler.Result);
             }
             catch (Exception e)
             {
                 _logger.LogError("Internal server error", e);
```

Comment:
Fixed, also added some more granularity in things that could go wrong in that method.